src/BaseList/BaseList.tsx

```diff
@@ -0,0 +1,153 @@
+import clsx from "clsx";
+import React from "react";
+import {
+  ListContext,
+  ListGridContext,
+  useGridStyles,
+  useListContext,
+} from "./context";
+import { ListGridTemplate, useStyles } from "./styles";
+
+export interface BaseListProps extends React.HTMLAttributes<HTMLElement> {
+  gridTemplate: ListGridTemplate;
+}
+export const BaseList: React.FC<BaseListProps> = ({
+  children,
+  gridTemplate,
+  ...props
+}) => (
+  <ListGridContext.Provider value={gridTemplate}>
+    <section {...props}>{children}</section>
+  </ListGridContext.Provider>
+);
+
+export type BaseListItemClassKey =
+  | "row"
+  | "rowBody"
+  | "rowHead"
+  | "rowFoot"
+  | "rowHover"
+  | "rowBodySelected";
+
+export interface BaseListItemProps
+  extends React.HTMLAttributes<HTMLDivElement | HTMLLIElement> {
+  classes: Record<BaseListItemClassKey, string>;
+  hover?: boolean;
+  selected?: boolean;
+}
+
+export const BaseListItem: React.FC<BaseListItemProps> = ({
+  classes,
+  className,
+  children,
+  hover = true,
+  selected,
+  ...props
+}) => {
+  const baseClasses = useStyles();
+  const gridClasses = useGridStyles();
+  const listSection = useListContext();
+
+  const Component = listSection === "body" ? "li" : "div";
+
+  return (
+    <Component
+      className={clsx(
+        className,
+        "OfsettedTableRow",
+        gridClasses.root,
+        classes.row,
+        baseClasses.row,
+        {
+          [classes.rowBody]: listSection === "body",
+          [classes.rowHead]: listSection === "head",
+          [classes.rowFoot]: listSection === "foot",
+          [classes.rowHover]: hover,
+          [classes.rowBodySelected]: selected,
+        }
+      )}
+      aria-selected={selected}
+      {...props}
+    >
+      {children}
+    </Component>
+  );
+};
+
+export type BaseListItemCellClassKey =
+  | "cell"
+  | "cellBody"
+  | "cellHeader"
+  | "cellAction"
+  | "cellCheckbox";
+
+export interface BaseListItemCellProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  classes: Record<BaseListItemCellClassKey, string>;
+  colSpan?: number;
+  padding?: "action" | "checkbox" | "none";
+}
+export const BaseListItemCell: React.FC<BaseListItemCellProps> = ({
```

Comment:
Why are there multiple components in one file, it's harder to navigate

Reply:
I can split it to 7 files containing 20 lines each, but I'm not sure if it actually resolves anything here. It's not like this file has 500 lines of random, uncoupled code.